drivers/sensor/tdk/icm40627/icm40627.c

```diff
@@ -0,0 +1,710 @@
+
+cleanup:
+	icm40627_unlock(dev);
+	return res;
+}
+
+static int icm40627_attr_set(const struct device *dev, enum sensor_channel chan,
+			     enum sensor_attribute attr, const struct sensor_value *val)
+{
+	int res = 0;
+
+	__ASSERT_NO_MSG(val != NULL);
+
+	icm40627_lock(dev);
+
+	switch (chan) {
+	case SENSOR_CHAN_ACCEL_X:
+	case SENSOR_CHAN_ACCEL_Y:
+	case SENSOR_CHAN_ACCEL_Z:
+	case SENSOR_CHAN_ACCEL_XYZ:
+		if (attr == SENSOR_ATTR_SAMPLING_FREQUENCY) {
+			res = icm40627_set_accel_odr(dev, val->val1);
+		} else if (attr == SENSOR_ATTR_FULL_SCALE) {
+			res = icm40627_set_accel_fs(dev, sensor_ms2_to_g(val));
+		} else {
+			LOG_ERR("Unsupported attribute");
+			res = -ENOTSUP;
+		}
+		break;
+
+	case SENSOR_CHAN_GYRO_X:
+	case SENSOR_CHAN_GYRO_Y:
+	case SENSOR_CHAN_GYRO_Z:
+	case SENSOR_CHAN_GYRO_XYZ:
+		if (attr == SENSOR_ATTR_SAMPLING_FREQUENCY) {
+			res = icm40627_set_gyro_odr(dev, val->val1);
+		} else if (attr == SENSOR_ATTR_FULL_SCALE) {
+			res = icm40627_set_gyro_fs(dev, sensor_rad_to_degrees(val));
+		} else {
+			LOG_ERR("Unsupported attribute");
+			res = -ENOTSUP;
+		}
+		break;
+
+	default:
+		LOG_ERR("Unsupported channel");
+		res = -ENOTSUP;
+		break;
+	}
+
+	icm40627_unlock(dev);
+
+	return res;
+}
+
+static int icm40627_attr_get(const struct device *dev, enum sensor_channel chan,
+			     enum sensor_attribute attr, struct sensor_value *val)
+{
+	const struct icm40627_data *data = dev->data;
+	int res = 0;
+
+	__ASSERT_NO_MSG(val != NULL);
+
+	icm40627_lock(dev);
+
+	val->val2 = 0;
+
+	switch (chan) {
+	case SENSOR_CHAN_ACCEL_X:
+	case SENSOR_CHAN_ACCEL_Y:
+	case SENSOR_CHAN_ACCEL_Z:
+	case SENSOR_CHAN_ACCEL_XYZ:
+		if (attr == SENSOR_ATTR_SAMPLING_FREQUENCY) {
+			val->val1 = data->accel_hz;
+		} else if (attr == SENSOR_ATTR_FULL_SCALE) {
+			val->val1 = data->accel_fs;
+		} else {
+			LOG_ERR("Unsupported attribute");
+			res = -ENOTSUP;
+		}
+		break;
+
+	case SENSOR_CHAN_GYRO_X:
+	case SENSOR_CHAN_GYRO_Y:
+	case SENSOR_CHAN_GYRO_Z:
+	case SENSOR_CHAN_GYRO_XYZ:
+		if (attr == SENSOR_ATTR_SAMPLING_FREQUENCY) {
+			val->val1 = data->gyro_hz;
+		} else if (attr == SENSOR_ATTR_FULL_SCALE) {
+			val->val1 = data->gyro_fs;
+		} else {
+			LOG_ERR("Unsupported attribute");
+			res = -ENOTSUP;
+		}
+		break;
+
+	default:
+		LOG_ERR("Unsupported channel");
+		res = -ENOTSUP;
+		break;
+	}
+
+	icm40627_unlock(dev);
+
+	return res;
+}
+
+static inline int icm40627_bus_check(const struct device *dev)
+{
+	const struct icm40627_config *cfg = dev->config;
+
+	return cfg->bus_io->check(&cfg->bus);
+}
+
+static int icm40627_init(const struct device *dev)
+{
+	if (icm40627_bus_check(dev) < 0) {
+		LOG_ERR("Bus is not ready");
+		return -ENODEV;
+	}
+
+	if (icm40627_sensor_init(dev)) {
+		LOG_ERR("Could not initialize sensor");
+		return -EIO;
+	}
+
+#ifdef CONFIG_ICM40627_TRIGGER
+	if (icm40627_trigger_init(dev)) {
+		LOG_ERR("Failed to initialize interrupts.");
+		return -EIO;
+	}
+#endif
+
+	int res = icm40627_turn_on_sensor(dev);
```

Comment:
Please move _turn_on_sensor() after enabling the interrupt in case of TRIGGER config